mule-extensions-api-xml-dsl/src/main/java/org/mule/runtime/extension/xml/dsl/api/resolver/DslElementResolver.java

```diff
@@ -349,25 +361,60 @@ private void declareFieldsAsChilds(final DslElementDeclarationBuilder objectBuil
 
     private boolean shouldGenerateChildElements(MetadataType metadataType, ExpressionSupport expressionSupport)
     {
-        boolean isExpressionRequired = ExpressionSupport.REQUIRED == expressionSupport;
-        boolean isPojo = metadataType instanceof ObjectType;
+        final AtomicBoolean supportsChildDeclaration = new AtomicBoolean(false);
 
-        Optional<ClassInformationAnnotation> classInformation = getSingleAnnotation(metadataType, ClassInformationAnnotation.class);
-        boolean isInstantiable = !isPojo || classInformation.map(ClassInformationAnnotation::isInstantiable).orElse(false);
-        boolean isExtensible = getSingleAnnotation(metadataType, ExtensibleTypeAnnotation.class).isPresent();
+        if (ExpressionSupport.REQUIRED == expressionSupport)
+        {
+            return false;
+        }
+
+        metadataType.accept(new MetadataTypeVisitor()
+        {
+            @Override
+            protected void defaultVisit(MetadataType metadataType)
+            {
+                supportsChildDeclaration.set(true);
+            }
+
+            @Override
+            public void visitArrayType(ArrayType arrayType)
+            {
+                arrayType.getType().accept(this);
+            }
+
+            @Override
+            public void visitObject(ObjectType objectType)
+            {
+                boolean isInstantiable = getSingleAnnotation(metadataType, ClassInformationAnnotation.class)
```

Comment:
Change IntrospectionUtils.isInstantiable to perform this logic, and fallback to the current logic if the annotation is not present. I don't want ClassInformationAnnotation to become widely used